### PR TITLE
tests: new spread test for netlink-audit interface

### PR DESCRIPTION
**Pull request by sergiocazzolato**

### CODE REVIEW

File: interfaces/builtin/netlink_audit.go [hunk 1]
```diff
@@ -35,14 +35,22 @@ bind
 socket AF_NETLINK - NETLINK_AUDIT
 `
 
+const netlinkAuditConnectedPlugAppArmor = `
+# Description: Can use netlink to read/write to kernel audit system.
+network netlink,
+# CAP_NET_ADMIN required per 'man 7 netlink'
+capability net_admin,
```
Comment (jdstrand) [inline interfaces/builtin/netlink_audit.go]: 'man 7 netlink' only mentions that NET_ADMIN is needed for multicast sockets and I think this comment should be made clear wrt that. Eg:
```
# CAP_NET_ADMIN required for multicast netlink sockets per 'man 7 netlink'
```

In addition, 'man 7 capabilties' says that you will also need 'capability audit_read'. I suggest:
```
# CAP_AUDIT_READ required to read the audit log via the netlink multicast socket
# per 'man 7 capabilities'
capability audit_read,
```

Note that CAP_AUDIT_READ was only added in 3.16.

Finally, can you separate these with a blank line? Eg:
```
# Description: Can use netlink to read/write to kernel audit system.
network netlink,

# CAP_NET_ADMIN required for multicast netlink sockets per 'man 7 netlink'
capability net_admin,

# CAP_AUDIT_READ required to read the audit log via the netlink multicast socket
# per 'man 7 capabilities'
capability audit_read,
```

File: tests/main/interfaces-netlink-connector/task.yaml [hunk 2]
```diff
@@ -12,14 +12,15 @@ prepare: |
     install_local test-snapd-netlink-connector
 
 execute: |
-    CONNECTED_PATTERN=":netlink-audit +test-snapd-netlink-connector"
+    CONNECTED_PATTERN=":netlink-connector +test-snapd-netlink-connector"
```
Comment (zyga) [inline tests/main/interfaces-netlink-connector/task.yaml]: This looks like an unrelated fix. If this branch doesn't come in, can you please cherry pick it and propose separately.

File: tests/main/interfaces-netlink-audit/task.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    The test-snapd-netlink-audit snap creates a netlink socket and binds it. 
+
+prepare: |
+    . $TESTSLIB/snaps.sh
+
+    # Install a snap declaring a plug on netlink-audit
+    install_local test-snapd-netlink-audit
+
+execute: |
+    CONNECTED_PATTERN=":netlink-audit +test-snapd-netlink-audit"
```
Comment (zyga) [inline tests/main/interfaces-netlink-audit/task.yaml]: Nitpick, why don't we put those into environment?

File: tests/lib/snaps/test-snapd-netlink-audit/bin/bind [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+if __name__ == "__main__":
+    # Create Netlink socket
+    # For more information about protocols see https://github.com/torvalds/linux/blob/master/include/uapi/linux/netlink.h
+    netlink_audit_protocol = 9
+    cn_idx_proc = 1
+
+    with socket.socket(socket.AF_NETLINK, socket.SOCK_DGRAM, netlink_audit_protocol) as sock:
+        sock.bind((os.getpid(), cn_idx_proc))
+        print('netlink connect bound')
```
Comment (zyga) [inline tests/lib/snaps/test-snapd-netlink-audit/bin/bind]: I think the print is stale, perhaps `netlink audit bound`?

File: tests/main/interfaces-netlink-audit/task.yaml (hunk 3 above)
Comment (sergiocazzolato) [reply tests/main/interfaces-netlink-audit/task.yaml]: done

File: tests/lib/snaps/test-snapd-netlink-audit/bin/bind (hunk 4 above)
Comment (sergiocazzolato) [reply tests/lib/snaps/test-snapd-netlink-audit/bin/bind]: done

File: tests/main/interfaces-netlink-connector/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/main/interfaces-netlink-connector/task.yaml]: sure

File: tests/main/interfaces-netlink-audit/task.yaml [hunk 5]
```diff
@@ -5,6 +5,10 @@ details: |
 
     The test-snapd-netlink-audit snap creates a netlink socket and binds it. 
 
+# Ubuntu 14.04-64 is skipped due to the capability CAP_AUDIT_READ
+# is not available in its kernel
+systems: [-ubuntu-14.04-*]
```
Comment (jdstrand) [inline tests/main/interfaces-netlink-audit/task.yaml]: I thought that Ubuntu 14.04 systems needed the lts kernel, so therefore 4.4. Is it not using the right kernel?

Comment (zyga) [reply tests/main/interfaces-netlink-audit/task.yaml]: Maybe one of the layers is out of date (libcap?). I agree with @jdstrand here, it should not be any different.
